apmpackage/cmd/gen-package/genpipelines.go

```diff
@@ -36,29 +93,40 @@ func generatePipelines(version, dataStream string) {
 		panic(err)
 	}
 
-	var definitions = make([]map[string]interface{}, 0)
+	var definitions = make([]PipelineDef, 0)
 	err = json.Unmarshal(bytes, &definitions)
 	if err != nil {
 		panic(err)
 	}
 
 	os.MkdirAll(pipelinesPath(version, dataStream), 0755)
 
+	var apmPipeline PipelineBody
 	for _, definition := range definitions {
-		pipeline, ok := definition["body"]
-		if !ok {
-			continue
-		}
-		id, ok := definition["id"]
-		if !ok {
+		pipeline := definition.Body
+		if definition.ID == "apm" {
+			apmPipeline = pipeline
 			continue
 		}
-
 		out, err := json.MarshalIndent(pipeline, "", "  ")
 		if err != nil {
 			panic(err)
 		}
-		fName := filepath.Join(pipelinesPath(version, dataStream), id.(string)+".json")
+		fName := filepath.Join(pipelinesPath(version, dataStream), definition.ID+".json")
 		ioutil.WriteFile(fName, out, 0644)
 	}
+
+	for _, p := range apmPipeline.Processors {
+		if p.Pipeline == nil {
+			// should not happen, lets panic loudly
+			panic(errors.New("expected pipeline processor"))
+		}
+		p.Pipeline.Name = streamMappings[dataStream] + "-" + version + "-" + p.Pipeline.Name
```

Comment:
Can you please add a comment as to why we're doing this? i.e. Fleet will rename it.

Reply:
```suggestion
		// name is updated to match the one generated by Fleet when installs the pipelines
		p.Pipeline.Name = streamMappings[dataStream] + "-" + version + "-" + p.Pipeline.Name
```